### ISSUE TITLE
Support screenshot for alert

### ISSUE THREAD

**Issue by arey** [FluentLenium/FluentLenium#455]
PNG and HTML screenshots silently failed on alert popup.
By debugging, the below line of the FluentDriver class:
```
final File scrFile = ((TakesScreenshot) getDriver()).getScreenshotAs(OutputType.FILE) 
```
throw an UnhandledAlertException exception:
```
Method threw 'org.openqa.selenium.UnhandledAlertException' exception.
```
To fix this limitation, there is some trick with Robot but I don't know if this solution is portable: http://stackoverflow.com/questions/21670361/capture-screenshot-of-alert

**Comment by filipcynarski** [comment]
Thank you @arey for adding this one to the issues. I think it can be fixed in next release.

**Comment by arey** [comment]
Thank you Filip

**Comment by Toilal** [comment]
I don't like the solution given on stackoverflow.com, because it won't work in many cases (remote driver, minified browser,...) . I think we could store the alert info, dismiss the alert, take the screenshot and use the java drawing API to display a fakee alertbox as an overlay on the generated screenshot. It won't be very sexy, but will do the job.

**Comment by filipcynarski** [comment]
@arey I'm seeing the issue is in the backlog from Dec - we haven't found good solution for the issue what I may suggest is to get the alert text instead and return it in exception message, we can also generate the screenshot which is a really fluentlenium generated message.

So summarising what we are going to achieve here is:
* Write test reproducing the issue
* Capture alert message and present it in error log
* Create our own image w/ the alert text but it is not a screenshot it is just our own image.

**Comment by arey** [comment]
Yes, your proposal suits me.

In parallel, do you have find or open a case into the bug tracker of the Selenium project?

**Comment by filipcynarski** [comment]
@arey nope because as I saw before Selenium says it is out of scope and is WAD, we will provide the solution soon

**Comment by filipcynarski** [comment]
code merged to develop
